pkg/syncer/metadatasyncer.go

```diff
@@ -337,12 +337,24 @@ func InitMetadataSyncer(ctx context.Context, clusterFlavor cnstypes.CnsClusterFl
 				volumeInfoCrDeletionMap[vcconfig.Host] = make(map[string]bool)
 				volumeOperationsLock[vcconfig.Host] = &sync.Mutex{}
 			}
-			// If it is a multi VC deployment, initialize volumeInfoService
-			if len(vcconfigs) > 1 && volumeInfoService == nil {
+			if clusterFlavor == cnstypes.CnsClusterFlavorVanilla {
+				// If it is a multi VC deployment, initialize volumeInfoService
+				if len(vcconfigs) > 1 && volumeInfoService == nil {
+					volumeInfoService, err = cnsvolumeinfo.InitVolumeInfoService(ctx)
+					if err != nil {
+						return logger.LogNewErrorf(log, "error initializing volumeInfoService. Error: %+v", err)
+					}
+				}
+			} else if clusterFlavor == cnstypes.CnsClusterFlavorWorkload &&
+				commonco.ContainerOrchestratorUtility.IsFSSEnabled(ctx, common.PodVMOnStretchedSupervisor) {
+				log.Info("Loading CnsVolumeInfo Service to persist mapping for VolumeID to StoragePolicyID")
```

Comment:
StoragePolicyId -> Storage policy info as you might store storageClassName as well